### PR TITLE
Add Vault System Backend Engine Policy and Kubernetes Auth method provisioning services

### PR DESCRIPTION
**Pull request by vsevel**
This provides provisioning services to create and configure:
 - vault policies
 - kubernetes auth method
 - kubernetes auth role

this may be used by clients that provision vault programmatically.
in my company we have developed a java program that is deploying a kubernetes application into the cluster, including the vault objects, such as the app kubernetes auth role and policy.

I have implemented all methods on https://www.vaultproject.io/api-docs/auth/kubernetes and https://www.vaultproject.io/api-docs/system/policy for the sake of completeness.

### CODE REVIEW

**Review by sberyozkin — APPROVED** [review]
@vsevel Hi Vincent, welcome back.

I honestly can't see what I can suggest to change, this PR just fits cleanly into the Quarkus Vault ecosystem of the interfaces/models,

`VaultSystemBackendEngine` is proving its worth too.

Does this PR fix some open issues ?

Nice work :-) I'll be happy to merge tomorrow, unless Guillaume @gsmet or Alex @lordofthejars will have some more comments

**Comment by vsevel** [comment]
@sberyozkin thanks :)

**Comment by lordofthejars** [comment]
:+1: I like the idea of having this, I review the code and I'll let you know

**Comment by lordofthejars** [comment]
Looks good to me. Just one question that I have not found in the PR is to make the Kubernetes auth backend enabled programmatically: `vault auth enable kubernetes`. I have not found any information on how to do it using Rest API in the Vault docs but in theory it should be possible.

But that's all, I'd prefer to have the `adoc` update in this PR as well, but suppose you are thinking on doing in another PR.

Thanks for this really nice improvement.

**Comment by vsevel** [comment]
> but in theory it should be possible.

I think this is the [sys/mounts](https://www.vaultproject.io/api-docs/system/mounts#enable-secrets-engine). but I have not tried it yet. I can do it in a follow up.

> But that's all, I'd prefer to have the `adoc` update in this PR as well, but suppose you are thinking on doing in another PR.

I will do it now in this PR.

thanks for your comments.

**Comment by vsevel** [comment]
I have added this [documentation](https://github.com/quarkusio/quarkus/pull/10887/files#diff-d82433cb99174139aa070d45b0dbd17f)

**Review by lordofthejars — APPROVED** [review]

**Comment by vsevel** [comment]
@gsmet documentation has been added, and the last commit has been squashed + rebased. if no other feedback, we should be good to go :)

File: extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultKubernetesAuthService.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package io.quarkus.vault;
+
+import java.util.List;
+
+import io.quarkus.vault.auth.VaultKubernetesAuthConfig;
+import io.quarkus.vault.auth.VaultKubernetesAuthRole;
+
+/**
+ * This services provides programmatic access to the Kubernetes auth method.
```
Comment (gastaldi) [inline extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultKubernetesAuthService.java:9]:
```suggestion
 * This service provides programmatic access to the Kubernetes auth method.
```

Comment (vsevel) [reply extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultKubernetesAuthService.java]: fixed and squashed. thanks for picking that up.

Comment (gastaldi) [reply extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultKubernetesAuthService.java]: Thank you! Gpod job! 😉

**Comment by vsevel** [comment]
@gastaldi the build failed on "rest data with panache". is there any way we can restart the build from where it failed? or should I force push a change to trigger a new build?

File: docs/src/main/asciidoc/vault.adoc [hunk 2]
```diff
@@ -443,6 +443,33 @@ The last relevant property is `quarkus.vault.tls.skip-verify`, which allows to c
 but without checking the certificate authenticity. This may be convenient in development, but is strongly
 discouraged in production as it is not more secure than talking to Vault in plain HTTP.
 
+== Vault Provisioning
+
+Beside the typical client use cases, the Quarkus extension can be used to provision Vault as well, 
+for instance as part of a CD pipeline. Specific endpoints support this scenario:
```
Comment (gsmet) [inline docs/src/main/asciidoc/vault.adoc:449]: Are they endpoints or CDI beans?

I think we should be clearer about how a user can make use of them.

**Review by gsmet — DISMISSED** [review]
Added a suggestion for the documentation.

File: docs/src/main/asciidoc/vault.adoc (hunk 2 above)
Comment (vsevel) [reply docs/src/main/asciidoc/vault.adoc]: hello. I have made it more explicit, plus I added an example.

**Comment by sberyozkin** [comment]
Looks ready to go now

**Review by gastaldi — APPROVED** [review]

**Comment by vsevel** [comment]
thanks

**Pull request merged**